src/wallet.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::{bail, Context, Result};
+use async_trait::async_trait;
+use http::header::CONTENT_TYPE;
+use tracing::warn;
+use url::Url;
+
+use crate::core::{
+    authorization_request::{
+        parameters::ResponseMode, verification::RequestVerifier, AuthorizationRequest,
+        AuthorizationRequestObject,
+    },
+    metadata::WalletMetadata,
+    response::{AuthorizationResponse, PostRedirection},
+    util::{base_request, HttpClient},
+};
+
+#[async_trait]
+pub trait Wallet: RequestVerifier + Sync {
+    type HttpClient: HttpClient + Send + Sync;
```

Comment:
All these `Send + Sync` will be a problem if we want to update vp_interop

Reply:
Discussed, we will leave it for now and address this later if needed.